Announcement Cog

Kitty-Therapist opened the pull request:
Finally got the announcement cog to work for this bot. 

It should allow you to grab a channel, and then uses the said channel to send an announcement as long that the bot has permissions.

I've tested it, and it seems to be working effectively.

Tarek1337 commented on ServerTools.py:
```diff
@@ -0,0 +1,40 @@
+import asyncio
+import discord
+from discord.ext import commands
+from discord.ext.commands import BadArgument, Greedy, MemberConverter, RoleConverter
```
Get rid of this line, you are not using any of these things.

Tarek1337 commented on ServerTools.py:
```diff
@@ -0,0 +1,40 @@
+from Bot import TheRealGearBot
+from Util import Permissioncheckers, Configuration, Utils, GearbotLogging, Pages, Emoji, Translator, \
+   Confirmation, MessageUtils
+from Util.Converters import DiscordUser, PotentialID, RoleMode, Guild, \
```
Thse are nto used either

Tarek1337 commented on ServerTools.py:
```diff
@@ -0,0 +1,40 @@
+
+import GearBot
+from Bot import TheRealGearBot
+from Util import Permissioncheckers, Configuration, Utils, GearbotLogging, Pages, Emoji, Translator, \
```
Only a few things are used here, remove the unused ones.

Kitty-Therapist commented:
Thanks. I've removed the unnecessary lines and adjusted them. As for the error, I was told by AEnterprise to make the string for the error where the bot didn't have SEND_MESSAGE permission. 

Lemme know if I missed a line that shouldn't be needed.

AEnterprise commented on ServerTools.py:
```diff
@@ -0,0 +1,36 @@
+    }
+    def __init__(self, bot):
+        self.bot: GearBot = bot
+        self.running = True
```
why are you keeping track on if it's running or a set for handling stuff this cog doesn't do any of that?

AEnterprise commented on lang/en_US.json:
```diff
@@ -547,5 +547,6 @@
   "emoji_name_space": "Emoji names can not have spaces",
-  "emoji_upload_rejected": "Discord refused the emoji, maybe it has a bad name or is corrupt?"
+  "emoji_upload_rejected": "Discord refused the emoji, maybe it has a bad name or is corrupt?",
+  "announce_invaild_id": "I was not able to talk in that channel. Are you sure that you have the correct channel and I have permission to view and talk in that particular channel?"
```
a bot can always 'see' a channel, if if you put read messages on explicitly denied, it still knows the channel is there, it just denies sending messages there

AEnterprise commented on ServerTools.py:
```diff
@@ -0,0 +1,36 @@
+    async def announce(self, ctx: commands.Context, channel:discord.TextChannel, *, message):
+        if message != None:
+            try:
+                await channel.send(f"{message}") 
```
little weird to put fstring here if you only use the var and nothing else in it?

AEnterprise commented on ServerTools.py:
```diff
@@ -0,0 +1,36 @@
+            try:
+                await channel.send(f"{message}") 
+            except discord.Forbidden:
+                await ctx.send(
```
can you use `MessageUtils.send_to(ctx, 'NO', 'announce_invaild_id')` instead? cleaner and i'm slowly working on routing all these types of messages through that central function

AEnterprise commented on ServerTools.py:
```diff
@@ -0,0 +1,36 @@
+        self.handling = set()
+
+    def __unload(self):
+        self.running = False
```
same as above, we're not using any background tasks, can just be removed

Kitty-Therapist commented:
Made the changes as per request. 👍

Tarek1337 commented on ServerTools.py:
```diff
@@ -0,0 +1,33 @@
+    async def announce(self, ctx: commands.Context, channel:discord.TextChannel, *, message):
+        if message != None:
+            try:
+                await channel.send("{message}") 
```
This will just send ``{message}`` to Discord, I don't think you want that.

Tarek1337 commented on ServerTools.py:
```diff
@@ -0,0 +1,33 @@
+
+import GearBot
+from Bot import TheRealGearBot
+from Util import Permissioncheckers, GearbotLogging, Emoji, Translator, MessageUtils
```
You still import stuff not used.

AEnterprise commented on ServerTools.py:
```diff
@@ -28,9 +25,9 @@ def __unload(self):
-                await ctx.send(
-                        f"{Emoji.get_chat_emoji('NO')} {Translator.translate('announce_invaild_id', ctx)}")
+                MessageUtils.send_to(ctx, 'NO', 'announce_invaild_id')
+                
```
this needs to be awaited

AEnterprise commented on ServerTools.py:
```diff
@@ -14,11 +14,8 @@ class ServerTools:
     def __init__(self, bot):
         self.bot: GearBot = bot
-        self.running = True
         self.handling = set()
```
this line also isn't needed

AEnterprise commented:
closing this due to age and that i have plans to do this more extensive on the dashboard instead
